mitiq/cdr/cdr_execution.py

```diff
@@ -0,0 +1,202 @@
+    Returns: The tuple (raw_expectations, mitigated_expectations)
+             corresponding to the many raw expectation values (at different
+             noise levels) and the associated mitigated values.
+
+    .. [Czarnik2020] : Piotr Czarnik, Andrew Arramsmith, Patrick Coles,
+        Lukasz Cincio, "Error mitigation with Clifford quantum circuit
+        data," (https://arxiv.org/abs/2005.10189).
+    .. [Lowe2020] : Angus Lowe, Max Hunter Gordon, Piotr Czarnik,
+        Andrew Arramsmith, Patrick Coles, Lukasz Cincio,
+        "Unified approach to data-driven error mitigation,"
+        (https://arxiv.org/abs/2011.01157)."""
+    # Extracting kwargs:
+    method_select = kwargs.get("method_select", "uniform")
+    method_replace = kwargs.get("method_replace", "closest")
+    random_state = kwargs.get("random_state", None)
+    training_set_generation_kwargs_keys = ["sigma_select", "sigma_replace"]
+    kwargs_for_training_set_generation = {
+        key: kwargs.get(key) for key in training_set_generation_kwargs_keys
+    }
+    training_circuits_list = generate_training_circuits(
+        circuit,
+        num_training_circuits,
+        fraction_non_clifford,
+        method_select,
+        method_replace,
+        random_state,
+        kwargs=kwargs_for_training_set_generation,
+    )
+    # If specified scale the noise in the circuit of interest and training
+    # circuits:
+    if scale_noise:
+        # to define number of paramters in defult linear model:
+        if not num_parameters:
+            num_parameters = len(scale_factors) + 1
+        training_circuits_list = scale_noise_in_circuits(
+            training_circuits_list, scale_noise, scale_factors
+        )
+        # redefine circuit as a list of circuits with scaled noise:
+        circuits = scale_noise_in_circuits(
+            [circuit], scale_noise, scale_factors
+        )
+    else:
+        # to define number of parameters in defult linear model:
+        if not num_parameters:
+            num_parameters = 2
+        # both need to be list of lists of circuits:
+        circuits = [[circuit]]
+        training_circuits_list = [training_circuits_list]
+    # execute the training circuits and the circuit of interest:
+    # list to store training circuits executed with hardware:
+    training_circuits_raw_data = [
+        [] for i in range(len(training_circuits_list))
+    ]
+    # list to store simulated training circuits:
+    training_circuits_simulated_data = []
+    for i, training_circuits in enumerate(training_circuits_list):
+        for j, circuit in enumerate(training_circuits):
+            training_circuits_raw_data[i].append(executor(circuit))
+            # runs the circuits with no increased noise in the simulator:
+            if i == 0:
+                training_circuits_simulated_data.append(simulator(circuit))
+    results_dict_training_circuits = [
+        training_circuits_raw_data,
+        training_circuits_simulated_data,
+    ]
```

Comment:
In `training_circuits_list` the circuits are ordered with increasing noise levels.
If so, isn't more natural placing the simulated data (zero noise) in the first position instead of last position? I.e.:
```python
results_dict_training_circuits = [
  training_circuits_simulated_data,
  training_circuits_raw_data,
]
```